procurement_purchase_no_grouping/models/stock_rule.py

```diff
@@ -15,9 +15,10 @@ class StockRule(models.Model):
 
     def _run_buy(self, procurements):
         for procurement, _rule in procurements:
-            grouping = procurement.product_id.categ_id.procured_purchase_grouping
-            if not grouping:
-                grouping = self.env.company.procured_purchase_grouping
+            grouping = (
+                procurement.product_id.categ_id.procured_purchase_grouping
+                or self.env.company.procured_purchase_grouping
+            )
             procurement.values["grouping"] = grouping
```

Comment:
And if you want to save a line in the code:

```suggestion
            procurement.values["grouping"] = (
                procurement.product_id.categ_id.procured_purchase_grouping
                or self.env.company.procured_purchase_grouping
            )